test/gtest/mhp/mhp-tests.cpp

```diff
@@ -4,6 +4,38 @@
 
 #include "mhp-tests.hpp"
 
+// Instantiate MHP-specific configurations for common tests
+template <typename T> struct CommonTestConfigBase {
+  using DV = mhp::distributed_vector<T>;
+  using DVA = mhp::distributed_vector<T, std::allocator<T>>;
+  using V = std::vector<T>;
+
+  static auto iota(auto &&r, auto val) { return mhp::iota(r, val); }
+  static auto for_each(auto &&policy, auto &&r, auto &&op) {
+    return mhp::for_each(policy, r, op);
+  }
+};
+template <typename T>
+struct CommonTestConfigCPU : public CommonTestConfigBase<T> {
+  using DV = mhp::distributed_vector<T>;
+  using DVA = mhp::distributed_vector<T, std::allocator<T>>;
+  static auto policy() { return std::execution::par_unseq; }
+};
+#ifdef SYCL_LANGUAGE_VERSION
+template <typename T>
+struct CommonTestConfigSYCL : public CommonTestConfigBase<T> {
+  using DV = mhp::distributed_vector<T, mhp::sycl_shared_allocator<T>>;
+  using DVA = DV;
+  static auto policy() { return mhp::device_policy(); }
+};
+#endif
+using Common_Types = ::testing::Types<
+#ifdef SYCL_LANGUAGE_VERSION
+    CommonTestConfigSYCL<int>, CommonTestConfigSYCL<float>,
```

Comment:
There is no special handling in the code for either 'int' or 'float' - the same code is executed. Code or branch coverage will be the same if just one is used, therefore there is no point in having `CommonTestConfigSYCL<int>` and `CommonTestConfigSYCL<float>`, but since we have to use some type with `CommonTestConfigSYCL` and `CommonTestConfigCPU` it is a good idea to use one with one type and another with another (best also having differents size).

Reply:
I don't think there is a lot of value, but there is some value in testing different types.

Reply:
It is enough to test vector of one particular size in UT unless it has some special handling depending on size.
For the same reason I also do not test templates with two types if handling them is identical.

But I don't insist - it's cosmetic.

Reply:
The fuzzing varies the size and alignment of subranges. It finds bugs, because distribution slices the ranges into segments and there are boundary conditions to get right. It would be good to write unit tests that accept parameters to get better coverage. I don't think gtest parameters will work for fuzzing, maybe global variables.

Reply:
My intuition is that fuzz tests are something different than unittests (unittests should be repeatable, non-random, simple, extremely fast). For quick search I see that google has separate framework for it https://github.com/google/fuzztest and there are some docs there about fuzz testing approaches, best practises, which I need to read about.

Reply:
google fuzztest looks good. It is a good match for the way we did fuzzing to satisfy SDL.

Fuzz tests are parameterized unit tests, also called property-based tests.
https://github.com/google/fuzztest/blob/main/doc/fuzz-test-macro.md

Not all tests need to be fuzzable, but we should fuzz test all the algorithms. It's another reason not to use initializer lists for inputs/results.